### ISSUE TITLE
`next build` doesn't terminate when `swcMinify: false` and `devtool: 'source-map'` are set

### ISSUE THREAD

**Issue by lforst** [vercel/next.js#45276]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: linux
      Arch: arm64
      Version: #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022
    Binaries:
      Node: 16.19.0
      npm: 8.19.3
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.1.6-canary.1
      eslint-config-next: 13.1.2
      react: 18.2.0
      react-dom: 18.2.0

### Link to the code that reproduces this issue

https://stackblitz.com/edit/vercel-next-js-shmwkm?file=next.config.js

### To Reproduce

1. Create Next.js config with the following:

```
module.exports = {
  reactStrictMode: true,
  swcMinify: false,
  webpack: (config) => {
    config.devtool = 'source-map';
    return config;
  },
};
```

2. Run `yarn build`
3. Examine the build being stuck on "Creating an optimized production build ..."

Side note: We noticed this because a lot of users were reporting that their app doesn't build anymore with `@sentry/nextjs` installed. Some more info with useful reports in this issue: https://github.com/getsentry/sentry-javascript/issues/6782#issuecomment-1403595116

### Describe the Bug

The build get stuck and doesn't terminate.

### Expected Behavior

Build succeeds.

**Comment by Max29a** [comment]
I am also having this problem and hoping very much for a solution

**Comment by 0xflumedev** [comment]
Waiting for a fix, major issue

**Comment by huozhi** [comment]
Can you try the latest canary `13.1.7-canary.0`? fixed by #45423
